File: prototype/rails/src/app/domain/orders/shipping_address_test.rb
# Runs without Rails: ruby -Iapp app/domain/orders/shipping_address_test.rb
require "minitest/autorun"
require_relative "shipping_address"

module Domain
  module Orders
    class ShippingAddressTest < Minitest::Test
      def test_it_carries_every_line_an_order_copies
        address = ShippingAddress.new(
          name: "Ada Lovelace", line1: "12 Analytical Way", line2: nil,
          city: "London", region: "Greater London", postal_code: "EC1A 1BB", country: "GB"
        )

        assert_equal(
          %i[name line1 line2 city region postal_code country],
          address.to_h.keys
        )
        assert_nil address.line2
      end
    end
  end
end

File: prototype/php/src/app/Domain/Orders/ShippingAddressTest.php
<?php

declare(strict_types=1);

namespace App\Domain\Orders;

it('names every column an order stores it in', function (): void {
    $address = ShippingAddress::to(
        name: 'Ada Lovelace',
        line1: '12 Analytical Way',
        line2: null,
        city: 'London',
        region: 'Greater London',
        postalCode: 'EC1A 1BB',
        country: 'GB',
    );

    expect($address->attributes())->toBe([
        'shipping_name' => 'Ada Lovelace',
        'shipping_line1' => '12 Analytical Way',
        'shipping_line2' => null,
        'shipping_city' => 'London',
        'shipping_region' => 'Greater London',
        'shipping_postal_code' => 'EC1A 1BB',
        'shipping_country' => 'GB',
    ]);
});
